In [1]:
import pandas as pd

In [2]:
titanic_data = pd.read_csv("titanic_data.csv")

In [3]:
titanic_data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

## Condition Based Filtering

In [6]:
mask = titanic_data["Fare"] > 100

In [7]:
fare_100  = titanic_data[mask]

In [9]:
fare_100.shape

(53, 12)

In [11]:
fare_100.head(10)

PassengerId  Survived  Pclass  \
27            28         0       1   
31            32         1       1   
88            89         1       1   
118          119         0       1   
195          196         1       1   
215          216         1       1   
258          259         1       1   
268          269         1       1   
269          270         1       1   
297          298         0       1   

                                               Name     Sex   Age  SibSp  \
27                   Fortune, Mr. Charles Alexander    male  19.0      3   
31   Spencer, Mrs. William Augustus (Marie Eugenie)  female   NaN      1   
88                       Fortune, Miss. Mabel Helen  female  23.0      3   
118                        Baxter, Mr. Quigg Edmond    male  24.0      0   
195                            Lurette, Miss. Elise  female  58.0      0   
215                         Newell, Miss. Madeleine  female  31.0      1   
258                                Ward, Miss. Anna  female  35.0      0   
268   Graham, Mrs. William Thompson (Edith Junkins)  female  58.0      0   
269                          Bissette, Miss. Amelia  female  35.0      0   
297                    Allison, Miss. Helen Loraine  female   2.0      1   

     Parch    Ticket      Fare        Cabin Embarked  
27       2     19950  263.0000  C23 C25 C27        S  
31       0  PC 17569  146.5208          B78        C  
88       2     19950  263.0000  C23 C25 C27        S  
118      1  PC 17558  247.5208      B58 B60        C  
195      0  PC 17569  146.5208          B80        C  
215      0     35273  113.2750          D36        C  
258      0  PC 17755  512.3292          NaN        C  
268      1  PC 17582  153.4625         C125        S  
269      0  PC 17760  135.6333          C99        S  
297      2    113781  151.5500      C22 C26        S

In [14]:
female_fare_100 = fare_100[fare_100.Sex == "female"]

In [20]:
female_fare_100.shape
## Why nearly 65% of highest paying customer are female in 1912 ?

(34, 12)

In [23]:
female_fare_100 = titanic_data[
    (titanic_data.Fare > 100) & (titanic_data.Sex == "female")
    ]
female_fare_100.head()

PassengerId  Survived  Pclass  \
31            32         1       1   
88            89         1       1   
195          196         1       1   
215          216         1       1   
258          259         1       1   

                                               Name     Sex   Age  SibSp  \
31   Spencer, Mrs. William Augustus (Marie Eugenie)  female   NaN      1   
88                       Fortune, Miss. Mabel Helen  female  23.0      3   
195                            Lurette, Miss. Elise  female  58.0      0   
215                         Newell, Miss. Madeleine  female  31.0      1   
258                                Ward, Miss. Anna  female  35.0      0   

     Parch    Ticket      Fare        Cabin Embarked  
31       0  PC 17569  146.5208          B78        C  
88       2     19950  263.0000  C23 C25 C27        S  
195      0  PC 17569  146.5208          B80        C  
215      0     35273  113.2750          D36        C  
258      0  PC 17755  512.3292          NaN        C

In [28]:
# Female survival Rate in Titanic

survived_female = titanic_data[
    (titanic_data['Sex'] == "female") & (titanic_data.Survived == 1)
]
survived_female.shape

(233, 12)

Out of 891 passengers , 233 female passengers survived in the crash.

In [30]:
survived_highclass = titanic_data[
    (titanic_data['Pclass'] == 1) & (titanic_data.Survived == 1)
]
survived_highclass.shape

(136, 12)

## Filtering using existing methods

In [34]:
# isna()
null_cabin = titanic_data[titanic_data.Cabin.isna()]
null_cabin.shape

(687, 12)

In [37]:
not_null_cabin = titanic_data[titanic_data.Cabin.notna()]
not_null_cabin.shape

(204, 12)

In [43]:
titanic_data[titanic_data['Ticket'].duplicated()]

PassengerId  Survived  Pclass  \
24            25         0       3   
71            72         0       3   
88            89         1       1   
117          118         0       2   
119          120         0       3   
..           ...       ...     ...   
876          877         0       3   
879          880         1       1   
880          881         1       2   
885          886         0       3   
888          889         0       3   

                                              Name     Sex   Age  SibSp  \
24                   Palsson, Miss. Torborg Danira  female   8.0      3   
71                      Goodwin, Miss. Lillian Amy  female  16.0      5   
88                      Fortune, Miss. Mabel Helen  female  23.0      3   
117                Turpin, Mr. William John Robert    male  29.0      1   
119              Andersson, Miss. Ellis Anna Maria  female   2.0      4   
..                                             ...     ...   ...    ...   
876                  Gustafsson, Mr. Alfred Ossian    male  20.0      0   
879  Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880   Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
885           Rice, Mrs. William (Margaret Norton)  female  39.0      0   
888       Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   

     Parch      Ticket      Fare        Cabin Embarked  
24       1      349909   21.0750          NaN        S  
71       2     CA 2144   46.9000          NaN        S  
88       2       19950  263.0000  C23 C25 C27        S  
117      0       11668   21.0000          NaN        S  
119      2      347082   31.2750          NaN        S  
..     ...         ...       ...          ...      ...  
876      0        7534    9.8458          NaN        S  
879      1       11767   83.1583          C50        C  
880      1      230433   26.0000          NaN        S  
885      5      382652   29.1250          NaN        Q  
888      2  W./C. 6607   23.4500          NaN        S  

[210 rows x 12 columns]

In [44]:
titanic_data[titanic_data.Ticket == "349909"]

PassengerId  Survived  Pclass  \
7              8         0       3   
24            25         0       3   
374          375         0       3   
567          568         0       3   

                                            Name     Sex   Age  SibSp  Parch  \
7                 Palsson, Master. Gosta Leonard    male   2.0      3      1   
24                 Palsson, Miss. Torborg Danira  female   8.0      3      1   
374                   Palsson, Miss. Stina Viola  female   3.0      3      1   
567  Palsson, Mrs. Nils (Alma Cornelia Berglund)  female  29.0      0      4   

     Ticket    Fare Cabin Embarked  
7    349909  21.075   NaN        S  
24   349909  21.075   NaN        S  
374  349909  21.075   NaN        S  
567  349909  21.075   NaN        S